### DISCUSSION TITLE
Missing ckpt file

### DISCUSSION THREAD

**Discussion by akindofyoga** [yoshitomo-matsubara/sc2-benchmark#109] [Q&A] [answered]
I tried running `python script/task/object_detection.py -test_only -student_only --config configs/coco2017/supervised_compression/entropic_student/faster_rcnn_splittable_resnet50-fp-beta0.08_fpn_from_faster_rcnn_resnet50_fpn.yaml --log log/coco2017/supervised_compression/entropic_student/faster_rcnn_splittable_resnet50-fp-beta0.08_fpn_from_faster_rcnn_resnet50_fpn.txt`

One of the lines that gets printed out says `INFO    torchdistill.common.main_util   ckpt file is not found at `./resource/ckpt/coco2017/teacher/coco2017-fasterrcnn_resnet50_fpn.pt``.

Is this a problem, or is it something that I can safely ignore? I plan to try running the code without the -test_only flag, and want to make sure I will be starting from existing weights. 

I have left all of the `pretrained: True` lines that are in the [original config file](https://github.com/yoshitomo-matsubara/sc2-benchmark/blob/main/configs/coco2017/supervised_compression/entropic_student/faster_rcnn_splittable_resnet50-fp-beta0.08_fpn_from_faster_rcnn_resnet50_fpn.yaml).

**Comment by yoshitomo-matsubara** [comment, marked as answer]
Hi @akindofyoga 

Unless you want to train a student model using your own checkpoint for a teacher, you can safely ignore the message.
All the configurations I am currently providing are using `pretrained: True` for teachers (i.e., no local checkpoints used for teachers)

**Reply by akindofyoga** [reply]
Thank you so much for the prompt reply!

I got the following results when running `python script/task/object_detection.py -test_only -student_only --config configs/coco2017/supervised_compression/entropic_student/faster_rcnn_splittable_resnet50-fp-beta0.08
_fpn_from_faster_rcnn_resnet50_fpn.yaml --log log/coco2017/supervised_compression/entropic_student/faster_rcnn_splittable_resnet50-fp-beta0.08_fpn_from_faster_rcnn_resnet50_fpn.txt`:
```
2022/10/06 21:30:31     INFO    torchdistill.common.main_util   Loading model parameters
2022/10/06 21:30:31     INFO    __main__        [Student: faster_rcnn_model]
2022/10/06 21:30:39     INFO    torchdistill.misc.log   Test:  [   0/5000]  eta: 6:19:08  model_time: 3.3354 (3.3354)  evaluator_time: 0.0647 (0.0647)  time: 4.5496  data: 1.1403  max mem: 8942
2022/10/06 21:39:57     INFO    torchdistill.misc.log   Test:  [1000/5000]  eta: 0:37:30  model_time: 0.4531 (0.5483)  evaluator_time: 0.0064 (0.0119)  time: 0.5533  data: 0.0002  max mem: 9167
2022/10/06 21:47:51     INFO    torchdistill.misc.log   Test:  [2000/5000]  eta: 0:25:53  model_time: 0.4308 (0.5026)  evaluator_time: 0.0066 (0.0134)  time: 0.4388  data: 0.0001  max mem: 9167
2022/10/06 21:55:19     INFO    torchdistill.misc.log   Test:  [3000/5000]  eta: 0:16:29  model_time: 0.4078 (0.4804)  evaluator_time: 0.0072 (0.0126)  time: 0.4281  data: 0.0001  max mem: 9167
2022/10/06 22:02:59     INFO    torchdistill.misc.log   Test:  [4000/5000]  eta: 0:08:06  model_time: 0.4495 (0.4722)  evaluator_time: 0.0064 (0.0123)  time: 0.4380  data: 0.0001  max mem: 9167
2022/10/06 22:10:33     INFO    torchdistill.misc.log   Test: Total time: 0:39:58
2022/10/06 22:10:33     INFO    __main__        Averaged stats: model_time: 0.4243 (0.4662)  evaluator_time: 0.0062 (0.0120)
2022/10/06 22:10:33     INFO    __main__        Accumulating evaluation results...
2022/10/06 22:10:41     INFO    __main__        DONE (t=8.23s).
2022/10/06 22:10:41     INFO    __main__        IoU metric: bbox
2022/10/06 22:10:41     INFO    __main__         Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.367
2022/10/06 22:10:41     INFO    __main__         Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.579
2022/10/06 22:10:41     INFO    __main__         Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.395
2022/10/06 22:10:41     INFO    __main__         Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.210
2022/10/06 22:10:41     INFO    __main__         Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.408
2022/10/06 22:10:41     INFO    __main__         Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.477
2022/10/06 22:10:41     INFO    __main__         Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.305
2022/10/06 22:10:41     INFO    __main__         Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.485
2022/10/06 22:10:41     INFO    __main__         Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.511
2022/10/06 22:10:41     INFO    __main__         Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.315
2022/10/06 22:10:41     INFO    __main__         Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.551
2022/10/06 22:10:41     INFO    __main__         Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.651
2022/10/06 22:10:42     INFO    sc2bench.analysis       Bottleneck size [KB]: mean 191.0359572265625 std 21.60045613156666 for 5000 samples
```

and these results when running `python script/task/object_detection.py -test_only -student_only --config configs/coco2017/supervised_compression/entropic_student/faster_rcnn_splittable_resnet50-fp-beta5.12_fpn_from_faster_rcnn_resnet50_fpn.yaml --log log/coco2017/supervised_compression/entropic_student/faster_rcnn_splittable_resnet50-fp-beta5.12_fpn_from_faster_rcnn_resnet50_fpn.txt-test_only`:
```
2022/10/06 22:33:08     INFO    torchdistill.common.main_util   Loading model parameters
2022/10/06 22:33:08     INFO    __main__        [Student: faster_rcnn_model]
2022/10/06 22:33:15     INFO    torchdistill.misc.log   Test:  [   0/5000]  eta: 5:54:43  model_time: 3.0159 (3.0159)  evaluator_time: 0.0395 (0.0395)  time: 4.2568  data: 1.1966  max mem: 8942
2022/10/06 22:43:05     INFO    torchdistill.misc.log   Test:  [1000/5000]  eta: 0:39:33  model_time: 0.4450 (0.5722)  evaluator_time: 0.0081 (0.0185)  time: 0.5541  data: 0.0002  max mem: 9167
2022/10/06 22:51:26     INFO    torchdistill.misc.log   Test:  [2000/5000]  eta: 0:27:21  model_time: 0.4232 (0.5289)  evaluator_time: 0.0090 (0.0163)  time: 0.4353  data: 0.0002  max mem: 9167
2022/10/06 22:59:08     INFO    torchdistill.misc.log   Test:  [3000/5000]  eta: 0:17:17  model_time: 0.4093 (0.5019)  evaluator_time: 0.0091 (0.0152)  time: 0.4296  data: 0.0001  max mem: 9167
2022/10/06 23:06:46     INFO    torchdistill.misc.log   Test:  [4000/5000]  eta: 0:08:23  model_time: 0.4459 (0.4874)  evaluator_time: 0.0076 (0.0146)  time: 0.4722  data: 0.0002  max mem: 9167
2022/10/06 23:14:22     INFO    torchdistill.misc.log   Test: Total time: 0:41:10
2022/10/06 23:14:22     INFO    __main__        Averaged stats: model_time: 0.4236 (0.4784)  evaluator_time: 0.0080 (0.0141)
2022/10/06 23:14:22     INFO    __main__        Accumulating evaluation results...
2022/10/06 23:14:32     INFO    __main__        DONE (t=9.91s).
2022/10/06 23:14:32     INFO    __main__        IoU metric: bbox
2022/10/06 23:14:32     INFO    __main__         Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.287
2022/10/06 23:14:32     INFO    __main__         Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.466
2022/10/06 23:14:32     INFO    __main__         Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.304
2022/10/06 23:14:32     INFO    __main__         Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.118
2022/10/06 23:14:32     INFO    __main__         Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.323
2022/10/06 23:14:32     INFO    __main__         Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.405
2022/10/06 23:14:32     INFO    __main__         Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.266
2022/10/06 23:14:32     INFO    __main__         Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.414
2022/10/06 23:14:32     INFO    __main__         Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.434
2022/10/06 23:14:32     INFO    __main__         Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.213
2022/10/06 23:14:32     INFO    __main__         Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.478
2022/10/06 23:14:32     INFO    __main__         Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.602
2022/10/06 23:14:32     INFO    sc2bench.analysis       Bottleneck size [KB]: mean 8.9430111328125 std 1.497765128506247 for 5000 samples
```

Was this in line with what you got? Or are my Average Precision  and Average Recall values lower?

**Reply by yoshitomo-matsubara** [reply]
Hi @akindofyoga 
Both the results (Bottleneck size and mAP) look correct to me :)

**Reply by akindofyoga** [reply]
Thank you so much! What does the beta in this for loop represent? https://github.com/yoshitomo-matsubara/sc2-benchmark/tree/main/script/task#23-entropic-student

I want to train an object detector to work on custom data. Which of the three approaches do you think would be easiest for me to use?

**Reply by yoshitomo-matsubara** [reply]
Hi @akindofyoga 

beta is a balancing factor between data size and model accuracy used during training.
The lower beta is, the higher the resulting model accuracy and data size are.
In [our SC2 paper](https://github.com/yoshitomo-matsubara/sc2-benchmark#citation), multiple beta values are used (for multiple experiments) to discuss tradeoff between data size and model accuracy (to be precise, size of encoder to be executed on a weaker device is also introduced for three-way tradeoff)

The easiest one in terms of implementation may be CR + BQ as you can control the tradeoff by channel size of convolution layers you design and train the model. But supervised compression like ours can significantly improve the tradeoff.

**Reply by akindofyoga** [reply]
If I wanted to use your supervised compression method, would I just have to train a Faster R-CNN model on your data and then use this as the teacher model while running your training script on the same data? Or is there more to it? 

I have a lot of experience running transfer learning, but I have never modified a model on any way.

**Reply by yoshitomo-matsubara** [reply]
If you train an object detector on your custom data (i.e., other than COCO 2017, in this case), you will first train a object detector (without any modifications in model) on the custom data. Then, use your trained model as a teacher to train a student model with entropy bottleneck like our student models on the custom data.
(In our study, student models are modified teacher models with entropy bottleneck so that student models can leverage teachers' learnt parameters for modules commonly used between teacher and student models)
